src/Simulation/QCTraceSimulator/DepthCounter.cs

```diff
@@ -48,21 +52,27 @@ public class Metrics
         {
             public const string Depth = "Depth";
             public const string StartTimeDifference = "StartTimeDifference";
+            public const string Width = "Width";
         }
 
-        public double[] StatisticsRecord(double Depth, double StartTimeDifference)
+        public double[] StatisticsRecord(double Depth, double StartTimeDifference, double Width)
         {
-            return new double[] { Depth, StartTimeDifference };
+            return new double[] { Depth, StartTimeDifference, Width };
         }
 
         #region IQCTraceSimulatorListener implementation 
         public object NewTracingData(long qubitId)
         {
-            return new QubitTimeMetrics();
+            return new QubitTimeMetrics(qubitId);
         }
 
         public void OnAllocate(object[] qubitsTraceData)
         {
+            QubitTimeMetrics[] qubitsMetrics = Utils.UnboxAs<QubitTimeMetrics>(qubitsTraceData);
+            foreach (QubitTimeMetrics metric in qubitsMetrics)
+            {
+                qubitAvailabilityTime.MarkQubitIdUsed(metric.QubitId);
+            }
```

Comment:
Is there a reason that `Utils.UnboxAs` is preferred to using `Enumerable.Cast`? E.g.:
```
            QubitTimeMetrics[] qubitsMetrics = qubitsTraceData.Cast<QubitTimeMetrics>().ToArray();
```
Relatedly, is an array needed at all here? The only usage is to immediately loop over its contents, such that you may be able to avoid allocating a new array:
```suggestion
            foreach (QubitTimeMetrics metric in qubitsTraceData.Cast<QubitTimeMetrics>())
            {
                qubitAvailabilityTime.MarkQubitIdUsed(metric.QubitId);
            }
```

Reply:
Just moved this code from elsewhere. Enumerable is a good improvement!